text/0000-fabric-private-chaincode.md

```diff
@@ -0,0 +1,284 @@
+ 
+* Step 2: Create the FPC Chaincode deployment artifact
+ 
+    One or more of the participants compiles an agreed-upon chaincode using the FPC compiler configuration. This links in the Chaincode Library, including its shim in the compiled package and creating a chaincode identifier called the MRENCLAVE. Details of this process will vary in future versions supporting other TEEs. Participants must be careful to use the same compiler configuration and version numbers of all included software. As long as they do, the MRENCLAVE will be the same for each compiled Chaincode Package.
+ 
+* Step 3: Install and Approve an FPC Chaincode
+ 
+    Administrators in each Org that will participate in the channel install the agreed-upon chaincode package on their FPC-enabled Peers following the standard Fabric 2.0 lifecycle process. They perform the installation of the chaincode tarball onto their Peers, receiving back a Package ID which includes a hash of the installed chaincode package. Each admin then sends a Chaincode Definition / Approve message containing the Package ID to the Orderer. This is an otherwise standard Chaincode Definition but is extended to include the MRENCLAVE identifier.
+ 
+    At this point, the originating admin waits until enough Approvals have been sent to the Orderer to fulfil the Endorsement Policy, then sends Commit Proposals to the Peers, and receiving their endorsements according to the standard Fabric process, submits them back in a final Chaincode Definition (extended to include the MRENCLAVE Identifier) with all the necessary Endorsements to the Orderer. The Peers receive back the endorsements from the Orderer and when they see that the endorsement policy is met, it is considered fully validated.
+ 
+* Step 4: Enclave Creation and Attestation Generation
+ 
+    Once their Peer has a fully validated chaincode, the admin performs a Setup transaction to the Library Chaincode's Go wrapper and thereby creates an new chaincode enclave on the endorsing peer and loaded the chaincode. In particular, when creating the enclave, the wrapper also fetches the MRENCLAVE and Channel ID of the Ledger Enclave and passes them with the create command. 
+    Now, having all necessary information, the new Enclave creates public and private signing and encryption key pairs and an Enclave Chaincode ID which is a hash of the Public Signing Key. Moreover, the new Chaincode Enclave binds to the trusted ledger enclave using the information as received during its creation.
```

Comment:
As I asked in the previous comment, I would like to understand how the keys are generated?

Reply:
See answer above.